### PR TITLE
XMP write to jpeg w/o xmp_toolkit

### PR DESCRIPTION
**Pull request by gpeacock**
Supports XMP write for JPEG without the xmp_write feature or xmp_toolkit.

## Checklist
- [ ] This PR represents a single feature, fix, or change.
- [ ] All applicable changes have been documented.
- [ ] Any `TO DO` items (or similar) have been entered as GitHub issues and the link to that issue has been included in a comment.

### CODE REVIEW

File: sdk/src/asset_handlers/jpeg_io.rs [hunk 1]
```diff
@@ -63,13 +63,46 @@ fn extract_xmp(seg: &JpegSegment) -> Option<String> {
...
+    input_stream.rewind()?;
+    input_stream.read_to_end(&mut buf).map_err(Error::IoError)?;
+    let mut jpeg = Jpeg::from_bytes(buf.into()).map_err(|_err| Error::EmbeddingError)?;
+
+    let segments = jpeg.segments_mut();
+    segments.retain(|seg| {
+        !(seg.marker() == markers::APP1 && seg.contents().starts_with(XMP_SIGNATURE))
+    });
+
+    if let Some(xmp) = xmp {
```
Comment (mauricefisher64) [inline sdk/src/asset_handlers/jpeg_io.rs:93]: Do we always want to just replace the XMP or should we merge with existing values?

File: sdk/src/asset_handlers/jpeg_io.rs [hunk 2]
```diff
@@ -538,37 +571,39 @@ impl AssetIO for JpegIO {
 }
 
 impl RemoteRefEmbed for JpegIO {
-    #[allow(unused_variables)]
+    //#[allow(unused_variables)]
```
Comment (mauricefisher64) [inline sdk/src/asset_handlers/jpeg_io.rs:574]: Go ahead and remove this line now

**Review by scouten-adobe — COMMENTED** [review]
I am :+1: once we address @mauricefisher64's question about replace vs merge.

File: sdk/src/asset_handlers/jpeg_io.rs (hunk 1 above)
Comment (bquinn) [reply sdk/src/asset_handlers/jpeg_io.rs:93]: Jumping in to a very old question, I would push for merging with existing values, given that many publishers will use both C2PA and IPTC (and Exif and other metadata) simultaneously to ensure maximum compatibility.

**Comment by gpeacock** [comment]
Already submitted on another PR

**Pull request closed without merging**